crates/sui-json-rpc/src/transaction_execution_api.rs

```diff
@@ -286,12 +286,28 @@ impl WriteApiServer for TransactionExecutionApi {
         sender_address: SuiAddress,
         tx_bytes: Base64,
         gas_price: Option<BigInt<u64>>,
-        _epoch: Option<BigInt<u64>>,
+        epoch: Option<BigInt<u64>>,
+        additional_args: Option<DevInspectArgs>,
     ) -> RpcResult<DevInspectResults> {
         with_tracing!(async move {
+            let DevInspectArgs {
+                gas_sponsor,
+                gas_budget,
+                gas_objects,
+                skip_checks,
+            } = additional_args.unwrap_or_default();
             let tx_kind: TransactionKind = self.convert_bytes(tx_bytes)?;
             self.state
-                .dev_inspect_transaction_block(sender_address, tx_kind, gas_price.map(|i| *i))
+                .dev_inspect_transaction_block(
+                    sender_address,
+                    tx_kind,
+                    gas_price.map(|i| *i),
+                    gas_budget.map(|i| *i),
```

Comment:
Could this be done with `.copied`?

```suggestion
                    gas_price.copied(),
                    gas_budget.copied(),
```

Reply:
The `map` here is for converting an `Option<BigInt<u64>>` to an `Option<u64>`.